sematic/ui/packages/main/src/hooks/logHooks.ts

```diff
@@ -129,20 +169,22 @@ export function useAccumulateLogsUntilEnd(hasMore: boolean,
             }
 
             setIsLoading(true);
-            const {pulledLines} = await latestGetNext.current(DiagnosticReasons.ACCUMULATE);
+            const { pulledLines, canceled } = await latestGetNext.current(DiagnosticReasons.ACCUMULATE);
 
             if (abortController.signal.aborted) {
                 break;
             }
             setIsLoading(false);
-            
-            // The server shouldn't return NaN, but just in case and be defensive,
-            // we don't want to add NaN to the accumulatedLines.
-            if (!isNaN(pulledLines)) {
-                accumulatedLines += pulledLines;
-                setAccumulatedLines(accumulatedLines);
-            }
 
+            if (canceled !== true) {
+                // The server shouldn't return NaN, but just in case and be defensive,
+                // we don't want to add NaN to the accumulatedLines.
+                if (!isNaN(pulledLines!)) {
```

Comment:
Also log if this happens?

Reply:
I interpret that we want to log the `isNaN` case instead of `!isNan`.  If so, a change is made as change.